x-pack/test/functional/apps/uptime/synthetics_integration.ts

```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const monitorName = 'Sample Synthetics integration';
+
+  const uptimePage = getPageObjects(['syntheticsIntegration']);
+  const testSubjects = getService('testSubjects');
+  const uptimeService = getService('uptime');
+
+  describe('When on the Synthetics Integration Policy Create Page', function () {
+    this.tags(['ciGroup6']);
+    const basicConfig = {
+      name: monitorName,
+      apmServiceName: 'Sample APM Service',
+      tags: 'sample tag',
+    };
+
+    describe('displays custom UI', () => {
+      before(async () => {
+        const version = await uptimeService.syntheticsPackage.getSyntheticsPackageVersion();
+        await uptimePage.syntheticsIntegration.navigateToPackagePage(version!);
+      });
+
+      it('should display policy view', async () => {
+        await testSubjects.existOrFail('monitorSettingsSection');
+      });
+
+      it('prevent saving when integration name, url/host, or schedule is missing', async () => {
+        const saveButton = await uptimePage.syntheticsIntegration.findSaveButton();
+        await saveButton.click();
+
+        await testSubjects.missingOrFail('packagePolicyCreateSuccessToast');
+      });
+    });
+
+    describe('create new policy', () => {
+      let version: string;
+      before(async () => {
+        await uptimeService.syntheticsPackage.deletePolicyByName('system-1');
+      });
+
+      beforeEach(async () => {
+        version = (await uptimeService.syntheticsPackage.getSyntheticsPackageVersion())!;
+        await uptimePage.syntheticsIntegration.navigateToPackagePage(version!);
+        await uptimeService.syntheticsPackage.deletePolicyByName(monitorName);
+      });
+
+      afterEach(async () => {
+        await uptimeService.syntheticsPackage.deletePolicyByName(monitorName);
+      });
+
+      it('allows saving when user enters a valid integration name and url/host', async () => {
+        // This test ensures that updates made to the Synthetics Policy are carried all the way through
+        // to the generated Agent Policy that is dispatch down to the Elastic Agent.
+        const config = {
```

Comment:
I'm noticing that the next few lines do tend to repeat a lot, with the only variable being the `'host'`/`'url'` config fields. We can probably simplify this file a bit by making a helper function we can call from each test that requires this exact logic.